### PR TITLE
Use pause manifest image

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:
As pause manifest code is merged part of https://github.com/kubernetes/kubernetes/pull/57723, now its time to remove all architecture-dependent pause imagename.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by ixdy** [comment]
/retest

Is there a minimum docker version needed for this to work properly?

**Comment by mkumatag** [comment]
@ixdy fat manifest pushed to docker via https://github.com/moby/moby/pull/18785 in docker `1.10.0` release, so minimum version we need is docker 1.10.0 release.

File: pkg/kubelet/dockershim/docker_sandbox.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ import (
 )
 
 const (
-	defaultSandboxImage = "gcr.io/google_containers/pause-amd64:3.1"
```
Comment (luxas) [inline pkg/kubelet/dockershim/docker_sandbox.go]: can this start use `test/utils/image` instead?

**Review by luxas — APPROVED** [review]
LGTM

File: pkg/kubelet/dockershim/docker_sandbox.go (hunk 1 above)
Comment (mkumatag) [reply pkg/kubelet/dockershim/docker_sandbox.go]: I'm little hesitant to use this one because it is completely outside of `pkg/` package.

Comment (ixdy) [reply pkg/kubelet/dockershim/docker_sandbox.go]: I agree with @mkumatag - I think we should avoid dependencies from `pkg/` -> `test/`.

**Review by luxas — APPROVED** [review]

**Comment by mkumatag** [comment]
Resolved the conflict and pushed it back, @luxas @ixdy PTAL.

File: test/utils/image/manifest.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ var (
 	NoSnatTestProxy          = ImageConfig{e2eRegistry, "no-snat-test-proxy", "1.0", true}
 	NWayHTTP                 = ImageConfig{e2eRegistry, "n-way-http", "1.0", true}
 	// When these values are updated, also update cmd/kubelet/app/options/container_runtime.go
```
Comment (ixdy) [inline test/utils/image/manifest.go]: per this comment, I think you need to update cmd/kubelet/app/options/container_runtime.go?

https://github.com/kubernetes/kubernetes/blob/6590ea6d5d50700d34255b1e037b2702ad26b7fc/cmd/kubelet/app/options/container_runtime.go#L36-L40

File: test/integration/framework/util.go [hunk 3]
```diff
@@ -35,25 +34,9 @@ const (
...
-		// The platform string may for example be "linux/amd64", "linux/arm" or "windows/amd64".
-		osArchArray := strings.Split(sVer.Platform, "/")
-		arch = osArchArray[1]
-	}
-	return arch
-}
-
-// GetPauseImageName fetches the pause image name for the same architecture as the apiserver.
+// GetPauseImageName returns the pause image name with proper version
 func GetPauseImageName(c clientset.Interface) string {
```
Comment (ixdy) [inline test/integration/framework/util.go]: minor not: his probably doesn't need to take `c` anymore.

Comment (mkumatag) [reply test/integration/framework/util.go]: after removing that c this function is dup of function from manifest.go file, so I consolidated all the functions into one and refactored the code a bit. PTAL at latest commits.

**Comment by mkumatag** [comment]
@vishh please help me getting this PR into this release!

**Comment by mkumatag** [comment]
@vishh can you PTAL at this PR?

**Comment by mkumatag** [comment]
@kubernetes/sig-node-pr-reviews can someone please help me reviewing this code?

**Comment by resouer** [comment]
Could you please squash the commits?

I will kindly ask @kubernetes/sig-node-pr-reviews for approve for you

**Comment by mkumatag** [comment]
@resouer done.! Thanks

**Comment by resouer** [comment]
/lgtm

@derekwaynecarr mind to approve this PR?

**Pull request merged**